### ISSUE TITLE
Incorrect Style/SafeNavigation autofix when a ! is used in conditional

### ISSUE THREAD

**Issue by A-Scott-Rowe** [bbatsov/rubocop#4745]
We are getting an incorrect autofix in a conditional when using a `!` 

`return unless user && !user.new_record? && user.valid?`
which correctly throws a `Style/SafeNavigation` autofixes to a non equivalent value.

--------

## Expected behavior

Not sure, maybe not auto fix?

## Actual behavior

it autofixes to `return unless !user&.new_record? && user.valid?` which I believe is non-equivalent

## Steps to reproduce the problem

This only happens when we use our own rubocop config. When running on standard config no errors are found.

```
AllCops:
  DisplayCopNames: true
  DisplayStyleGuide: true
  TargetRubyVersion: 2.3
  TargetRailsVersion: 4.2
Rails:
  Enabled: true
Rails/SkipsModelValidations:
  Exclude:
    - spec/**/*
Encoding:
  Enabled: false
Metrics/LineLength:
  Max: 180
MethodLength:
  Max: 40
Metrics/BlockLength:
  Enabled: false
Documentation:
  Enabled: false
CyclomaticComplexity:
  Enabled: false
ClassAndModuleChildren:
  Enabled: false
Style/FileName:
  Enabled: false
Style/ClassAndModuleChildren:
  SupportedStyles: compact
Style/AccessorMethodName:
  Enabled: false
ClassLength:
  Max: 210
PerceivedComplexity:
  Enabled: false
TrivialAccessors:
  Enabled: false
Lint/RescueException:
  Enabled: false
RedundantSelf:
  Enabled: false
Layout/AlignParameters:
  Enabled: false
Style/Lambda:
  Enabled: false
AbcSize:
  Enabled: false
Style/BlockDelimiters:
  Enabled: true
  Exclude:
    - spec/**/*_spec.rb # For .change blocks.
MultilineOperationIndentation:
  Enabled: false
Security/YAMLLoad:
  Enabled: false
Lint/EndAlignment:
  EnforcedStyleAlignWith: variable
Lint/AmbiguousBlockAssociation:
  Enabled: false
Style/MixinGrouping:
  Enabled: false

# The following should be re-enabled for ruby 3 update
Style/StringLiterals:
  Enabled: false
Style/ConditionalAssignment:
  Enabled: false
Style/MutableConstant:
  Enabled: false
Performance/TimesMap:
  Enabled: false
Style/IfInsideElse:
  Enabled: false
Style/FormatStringToken:
  Enabled: false

# These only apply to rails 5 and above
Rails/HttpPositionalArguments:
  Enabled: false
Style/FrozenStringLiteralComment:
  Enabled: false

```

so using that config and the code 

```
# this is an aweesome class
class RubocopTest
  def test(user)
    return unless user && !user.new_record? && user.valid?
  end
end
```

run `rubocop -c <config_above> -a <rubocop_test.rb>`

## RuboCop version

```
$ rubocop -V
0.50.0 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin15)
```

**Comment by basex** [comment]
Made some tests, this commit is creating problems:

is replacing:
```ruby
return if user && !user.blank?
```

to

```ruby
return if !user.blank?
```

If `user`is `NIL`, on the first base the `if` returns `true`, and on the second case the `if` returns `false`.
